library/spdm_common_lib/context_data.c

```diff
@@ -129,26 +129,6 @@ return_status spdm_set_data(IN void *context, IN spdm_data_type_t data_type,
 		if (data_size != sizeof(uint32)) {
 			return RETURN_INVALID_PARAMETER;
 		}
-

```

Comment:
@steven-bellock 
Responder and requester both use spdm_set_data() to set capability field, it will be a bug that check responder related info when setting requester capability.
I moved the checked code to the responder init function.